iothub/device/tests/DeviceClientTwinApiTests.cs

```diff
@@ -151,14 +154,17 @@ public async Task DeviceClientUpdateReportedPropertiesAsyncCallsSendTwinPatchAsy
             var client = DeviceClient.CreateFromConnectionString(fakeConnectionString);
             client.InnerHandler = innerHandler;
             var props = new TwinCollection();
+            string body = JsonConvert.SerializeObject(props);
 
             // act
             await client.UpdateReportedPropertiesAsync(props).ConfigureAwait(false);
 
             // assert
             await innerHandler.
                 Received(1).
-                SendTwinPatchAsync(Arg.Is(props), Arg.Any<CancellationToken>()).ConfigureAwait(false);
+                SendClientTwinPropertyPatchAsync(
+                    Arg.Any<Stream>(),
+                    Arg.Any<CancellationToken>()).ConfigureAwait(false);
```

Comment:
Cannot figure out why the following doesn't work; any ideas?
```csharp
await innerHandler.
    Received(1).
    SendClientTwinPropertyPatchAsync(
        Arg.Is<Stream>(stream => new StreamReader(stream, Encoding.UTF8).ReadToEnd() == body),
        Arg.Any<CancellationToken>()).ConfigureAwait(false);
```

Reply:
Perhaps just create a mock for this method, capture the body in a method-local variable, and then assert on it later. If you need to validate the call count, you can increment an integer.